drivers/led/Kconfig.lp50xx

```diff
@@ -1,11 +1,51 @@
+# Copyright (c) 2023 Kistler Instrumente AG
 # Copyright (c) 2020 Seagate Technology LLC
 # SPDX-License-Identifier: Apache-2.0
 
 config LP50XX
 	bool "LP50xx LED controller"
 	default y
-	depends on DT_HAS_TI_LP503X_ENABLED
+	depends on DT_HAS_TI_LP503X_ENABLED || DT_HAS_TI_LP5009_ENABLED || \
+		   DT_HAS_TI_LP5012_ENABLED || DT_HAS_TI_LP5018_ENABLED || \
+		   DT_HAS_TI_LP5024_ENABLED
 	select I2C
 	help
-	  Enable driver for the Texas Instruments LP5030 and LP5036 I2C LED
-	  controllers. They are respectively supporting up to 10 and 12 LEDs.
+	  Enable driver for the Texas Instruments LP5009, LP5012, LP5018, LP5024, LP5030 and LP5036
+	  I2C LED controllers. They are respectively supporting 3 to 12 (RGB) LEDs.
+
+
+if LP50XX
+
+config LP50XX_ALLOW_SOFTWARE_BLINK
+	bool "Use software PWM for blink"
+	default n
+	help
+	  Enable software timer for LED API blink. Disable it to save memory
+	  and CPU resources.
+
+endif # DT_HAS_TI_LP5018_ENABLED
+
+if LP50XX_ALLOW_SOFTWARE_BLINK
```

Comment:
This commit should be split into several: one or more to add controller series support and one commit per feature you want to add (like software blinking).

About software blinking... It is not something we want to implement at the driver level. We want end-driver to expose hardware capabilities through the LED API. If a controller can blink its LEDs, then led_api_blink() should be implemented. If not, then it should be let empty. Software blinking should be considered as a fallback if hardware support is missing. This way, generic software blinking can be implemented in LED core and shared between all the LED drivers.

Please have a look at https://github.com/zephyrproject-rtos/zephyr/pull/50511. The same arguments about color correction have been developed.

Reply:
Just some thoughts:

> This commit should be split into several ...

Okay, let's wait for progress in #46134

> About software blinking... 

Are there any plans to do a LED core with software features which are not supported by the driver(s)?
Otherwise we could let the user decide via Kconfig as in this PR. What is your opinion regarding Kconfig?

Reply:
Hi @dariis. My preference is to have software LED effects implemented in some LED core code and used as a fall-back when there is no hardware support available (i.e. when the end-driver API function is missing). I am not working on it yet but it is in my todo list.

Reply:
For a start, yes. Eventually you also have to split the main commit (if possible). If you want to continue with this PR then I'll start the review process at the next iteration.

Reply:
> Hi @dariis. My preference is to have software LED effects implemented in some LED core code and used as fall-backs when there is no hardware support available (i.e. when the end-driver API function is missing). I am not working on it yet but it is in my todo list.

Hi is there any progress on this?

Reply:
> > Hi @dariis. My preference is to have software LED effects implemented in some LED core code and used as fall-backs when there is no hardware support available (i.e. when the end-driver API function is missing). I am not working on it yet but it is in my todo list.
> 
> Hi is there any progress on this?

No, it is still on my to-do list.